test/screenshot-test.js

```diff
@@ -44,65 +44,105 @@ describe("screentshot.capture", function () {
-        const actualArgs = writeFileSyncStub.getCall(0).args;
-        expect(file).to.be.equal(screenShotFile);
-        expect(actualArgs).to.be.deep.equal(expectedArgs);
-        done();
+        screenshot.capture().then(function (file) {
+          const filePath = path.join(screenshotsDir, screenShotFile);
+          const expectedArgs = [filePath, Buffer.from("screentshot").toString("base64"), "base64"];
+          const actualArgs = writeFileSyncStub.getCall(0).args;
+          expect(file).to.be.equal(screenShotFile);
+          expect(actualArgs).to.be.deep.equal(expectedArgs);
+          done();
+        });
       });
-    });
 
-    it("Should capture screentshot with sync function", function (done) {
-      sandbox.stub(process.hrtime, "bigint").returns(6767787989089);
-      const screenShotFile = "screenshot-6767787989089.png";
-      global.gauge = { screenshotFn: screenshotFunction };
-      const writeFileSyncStub = sandbox.stub(fs, "writeFileSync");
+      it("Should capture screentshot with sync function", function (done) {
+        sandbox.stub(process.hrtime, "bigint").returns(6767787989089);
+        const screenShotFile = "screenshot-6767787989089.png";
+        global.gauge = { screenshotFn: screenshotFunction };
+        const writeFileSyncStub = sandbox.stub(fs, "writeFileSync");
 
-      screenshot.capture().then(function (file) {
-        const filePath = path.join(screenshotsDir, screenShotFile);
-        const expectedArgs = [filePath, Buffer.from("screentshot").toString("base64")];
-        const actualArgs = writeFileSyncStub.getCall(0).args;
-        expect(file).to.be.equal(screenShotFile);
-        expect(actualArgs).to.be.deep.equal(expectedArgs);
-        done();
+        screenshot.capture().then(function (file) {
+          const filePath = path.join(screenshotsDir, screenShotFile);
+          const expectedArgs = [filePath, Buffer.from("screentshot").toString("base64"), "base64"];
+          const actualArgs = writeFileSyncStub.getCall(0).args;
+          expect(file).to.be.equal(screenShotFile);
+          expect(actualArgs).to.be.deep.equal(expectedArgs);
+          done();
+        });
       });
-    });
-  });
 
-  describe("with custom screenshot writer function", () => {
-    afterEach(() => {
-      global.gauge = { customScreenshotWriter: null };
-    });
-    it("Should capture screentshot with async function", function (done) {
-      const screenShotFile = "screenshot-file-1.png";
-      global.gauge = { customScreenshotWriter: function() {
-        return Promise.resolve(screenShotFile);
-      } };
+      describe("when data is in bytes", () => {
+        it("Should capture screentshot with async function", function (done) {
+          sandbox.stub(process.hrtime, "bigint").returns(6767787989089);
+          const screenShotFile = "screenshot-6767787989089.png";
+          global.gauge = { screenshotFn: () => Promise.resolve(Buffer.from("screentshot")) };
```

Comment:
should use `async` / `await` in the syntax or rename the spec to mention "function returning a promise"